src/store/adjustVault.ts

```diff
@@ -0,0 +1,231 @@
+import { AmountMath } from '@agoric/ertp';
+import { Amount } from '@agoric/ertp/src/types';
+import { calculateCurrentDebt } from '@agoric/inter-protocol/src/interest-math';
+import {
+  addRatios,
+  ceilMultiplyBy,
+  makeRatioFromAmounts,
+} from '@agoric/zoe/src/contractSupport';
+import { ratioGTE } from '@agoric/zoe/src/contractSupport/ratio';
+import { atom } from 'jotai';
+import { pursesAtom } from './app';
+import { vaultKeyToAdjustAtom, vaultStoreAtom } from './vaults';
+
+export const vaultToAdjustAtom = atom(get => {
+  const key = get(vaultKeyToAdjustAtom);
+  const { vaults, vaultManagers, prices, vaultGovernedParams, vaultMetrics } =
+    get(vaultStoreAtom);
+
+  const vault = key && vaults?.get(key);
```

Comment:
I think this came up in a previous PR. Why check for `key` value when `.get` will return the correct value regardless?
```
const vault = vaults?.get(key);
```
Is the above a type checker error?

Reply:
`key` is nullable, `vaults` requires a non-nullable key